sql/core/src/test/resources/sql-tests/results/join-lateral.sql.out

```diff
@@ -317,11 +317,18 @@ SELECT * FROM t1, LATERAL (SELECT c1 + c2 + rand(0) AS c3)
 struct<>
 -- !query output
 org.apache.spark.sql.AnalysisException
-Non-deterministic lateral subqueries are not supported when joining with outer relations that produce more than one row
```

Comment:
Should we set `spark.sql.error.includePlans` to `true` when generating this golden file?

Reply:
Maybe we should output the canonicalized plan string instead of using `toString`?

Reply:
Hi @allisonwang-db  I looked at this and the existing `plan.canonicalized` uses `transformUp` which unfortunately is not allowed during analysis. However I was able to implement this by manually canonicalizing just the expression IDs for purposes of these strings and now the tests should be deterministic. Thanks for the suggestion.